src/tools/wasm-split/instrumenter.cpp

```diff
@@ -62,6 +65,22 @@ void Instrumenter::addGlobals() {
   }
 }
 
+void Instrumenter::addSecondaryMemory(size_t numFuncs) {
+  if (options.storageKind != WasmSplitOptions::StorageKind::InSecondaryMemory) {
+    // Don't need secondary memory
+    return;
+  }
+  if (!wasm->features.hasMultiMemories()) {
+    Fatal()
+      << "error: --in-secondary-memory requires multi-memories to be enabled";
+  }
+
+  secondaryMemory = Names::getValidMemoryName(*wasm, "split_data");
+  // Create a memory with enough pages to write into
+  size_t pages = (numFuncs + Memory::kPageSize - 1) / Memory::kPageSize;
+  wasm->addMemory(Builder::makeMemory(secondaryMemory, pages, pages, true));
```

Comment:
Here's how you can make the secondary module imported. We should choose some reasonable default for the import module and base name, but ideally make them each configurable via the command line.

```suggestion
  auto mem = Builder::makeMemory(secondaryMemory, pages, pages, true)
  mem->module = "???";
  mem->base = "???";
  wasm->addMemory(std::move(mem));
```